### PR TITLE
A new configuration which assign the position of an operation

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Fixes https://github.com/Azure/autorest.csharp/issues/1650

# Description

This PR adds a new mgmt configuration to assign an operation to its resource class, collection class, or both.
```
operation-positions:
   /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.EdgeOrder/orders: collection
```

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: samples/Azure.Management.Storage/readme.md [hunk 1]
```diff
@@ -17,9 +17,19 @@ modelerfour:
...
-- /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/managementPolicies/{managementPolicyName}
+request-path-to-singleton-resource:
+  /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/managementPolicies/{managementPolicyName}: managementPolicies/default
+
+request-path-to-parent:
+  /subscriptions/{subscriptionId}/providers/Microsoft.Storage/deletedAccounts: /subscriptions/{subscriptionId}/providers/Microsoft.Storage/locations/{location}/deletedAccounts/{deletedAccountName}
+  /subscriptions/{subscriptionId}/providers/Microsoft.Storage/checkNameAvailability: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}
+
+operation-positions:
+  /subscriptions/{subscriptionId}/providers/Microsoft.Storage/deletedAccounts: collection
```
Comment (AlexanderSher) [inline samples/Azure.Management.Storage/readme.md:28]: What would happen if the operation on this path wasn't found?

File: src/AutoRest.CSharp/Mgmt/Output/Resource.cs [hunk 2]
```diff
@@ -18,8 +18,13 @@ namespace AutoRest.CSharp.Mgmt.Output
 {
     internal class Resource : MgmtTypeProvider
     {
+        protected static readonly string ResourcePosition = "resource";
+        protected static readonly string CollectionPosition = "collection";
+
         private static readonly HttpMethod[] MethodToExclude = new[] { HttpMethod.Put, HttpMethod.Get, HttpMethod.Delete, HttpMethod.Patch };
 
+        protected virtual string Position => ResourcePosition;
```
Comment (AlexanderSher) [inline src/AutoRest.CSharp/Mgmt/Output/Resource.cs:26]: Can we assign the value in constructor instead of making property virtual?

Comment (ArcturusZhang) [reply src/AutoRest.CSharp/Mgmt/Output/Resource.cs:26]: Make sense

File: samples/Azure.Management.Storage/readme.md (hunk 1 above)
Comment (ArcturusZhang) [reply samples/Azure.Management.Storage/readme.md:28]: Unfortunately we currently do not have a validation of the paths in the configurations, therefore if this path does not exist, nothing will happen as if this configuration is not here.

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**